plugins/code-coverage/src/components/CoverageHistoryChart/CoverageHistoryChart.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Box,
+  Card,
+  CardContent,
+  MenuItem,
+  Select,
+  Typography,
+} from '@material-ui/core';
+import {
+  LineChart,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  Line,
+  CartesianGrid,
+  ResponsiveContainer,
+} from 'recharts';
+import { useAsync } from 'react-use';
+import { useApi } from '@backstage/core-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { codeCoverageApiRef } from '../../api';
+import { Progress } from '@backstage/core';
+import { Alert } from '@material-ui/lab';
+import TrendingDownIcon from '@material-ui/icons/TrendingDown';
+import TrendingUpIcon from '@material-ui/icons/TrendingUp';
+import TrendingFlatIcon from '@material-ui/icons/TrendingFlat';
+
+type Coverage = 'line' | 'branch';
+
+const getTrendIcon = (trend: number) => {
+  switch (true) {
```

Comment:
TIL

Reply:
this is maybe not good practice and should be if statements, but i figured since the cases were only dealing with one variable and pretty simple that this was a bit nicer to look at :)